src/epics/data.ts

```diff
@@ -130,6 +130,18 @@ export const bcFetchDataEpic: Epic = (action$, store) =>
                 .fetchBcData(state.screen.screenName, bcUrl, fetchParams, canceler.cancelToken)
                 .mergeMap(data => {
                     const newCursor = data.data[0]?.id
+                    const changeCurrentCursor = Observable.of<AnyAction>(
+                        $do.bcChangeCursors({
+                            cursorsMap: {
+                                [bcName]: data.data.length > 0 && cursor && data.data.some(i => i.id === cursor) ? cursor : newCursor
```

Comment:
```
[bcName]: data.data?.some(i => i.id === cursor) ? cursor : newCursor
```
`?.some` already handles nullable in data response so no need to check for length explicitly; also if we have a record in our data with id `undefined` then we screwed too much for the check `cursor &&` to matter

Reply:
Condition is changed to `data.data.some(i => i.id === cursor)`